packages/react-router-dom/index.tsx

```diff
@@ -403,40 +413,50 @@ export const NavLink = React.forwardRef<HTMLAnchorElement, NavLinkProps>(
     },
     ref
   ) {
-    let location = useLocation();
     let path = useResolvedPath(to);
+    let match = useMatch({ path: path.pathname, end, caseSensitive });
+
+    let routerState = React.useContext(UNSAFE_DataRouterStateContext);
+    let nextLocation = routerState?.navigation.location;
```

Comment:
Total side note, but we should make a hook for accessicing DataRouterStateContext that uses an `invariant` to warn if there is no context (they rendered the hook outside of a data router) and then return a type w/o the `| null` junk so we don't need these checks.

Reply:
haha, now that I go looking, I actually have a private one of those used inside the other hooks that I can grab.

Reply:
Actually, scratch that.  We're doing what you're getting at in our data-router-only hooks (`useLoaderData` etc.).  This is a slightly different use-case since `NavLink` is allowed to be used outside a `DataRouter`, and when it is - `isPending` is just always `false`.  So we don't want an `invariant` if the context doesn't exist here